### DISCUSSION TITLE
Deep-Q Learning demo / example

### DISCUSSION THREAD

**Discussion by dwctic** [deepjavalibrary/djl#3481] [Development]
The only example using the QAgent is the tic-tac-toe one, and after playing with it, It doesnt appear that it actually trains a worthwhile model. After running some simulations it doesn't even appear to have learned how to block the opposing player when a win is emminent. I don't mind putting in the effort to update the example, but I need to better understand how (or if) the QAgent api even works, and am hoping someone else who has used it can help.

Is the agent training against itself only? If so, could this be part of the problem? Even when the validation win rate is 90% the model still doesn't seem to predict optimal moves.

**Comment by frankfliu** [comment]
You might want to take a look this example: https://towardsdatascience.com/train-undying-flappy-bird-using-reinforcement-learning-on-java-98ff68eb28bf
